core/src/main/antlr4/com/databricks/labs/remorph/parsers/snowflake/SnowflakeLexer.g4

```diff
@@ -914,4 +914,28 @@ fragment DEC_DIGIT   : [0-9];
 // This lexer rule is needed so that any unknown character in the lexicon does not
 // cause an incomprehensible error message. This rule will allow the parser to issue
 // somethign more meaningful and perform error recovery.
-BADCHAR: .;
\ No newline at end of file
+BADCHAR: .;
+
+WS: [ \t\n\r\f]+ -> skip;
+SQLCOMMAND:
+    '!' WS* (
+        'abort'
+        | 'connect'
+        | 'define'
+        | 'edit'
+        | 'exit'
+        | 'help'
+        | 'options'
+        | 'pause'
+        | 'print'
+        | 'queries'
+        | 'quit'
+        | 'rehash'
+        | 'result'
+        | 'set'
+        | 'source'
+        | 'spool'
+        | 'system'
+        | 'variables'
+    ) ~[\r\n]*
```

Comment:
I think that this will fail if the input does not have a trailing \n, which can happen on Windows, or tests. This should work though:

```antlr
(~[\r\n] | EOF)*
```

Write a test where the input is just a single `!` with no `\n`, a test where it is `!set` and a test that is `! badcommand` (and checks that a syntax error is thrown'. See `TSqlErrorStategySpec` for testing for syntax errors.

Reply:
@jimidle as disscussed this will work fine and no further change needed.